hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -479,7 +480,11 @@ def test_function(self, data: ConjectureData) -> None:
         except BackendCannotProceed as exc:
             if exc.scope in ("verified", "exhausted"):
                 self._switch_to_hypothesis_provider = True
-                if exc.scope == "verified":
+                if exc.scope == "verified" and not self._backend_found_failure:
+                    # A backend might report a failure and then report verified
+                    # afterwards, which is to be interpreted only as "there are
+                    # no more failures *other than what we already reported*".
+                    # Do not report this as unsound.
                     self._verified_by = self.settings.backend
```

Comment:
Hmm, should we set `_verified_by` and then make the `_backend_found_failure` check where we'd raise the unsoundness error?  That'd make the comment somewhat more obvious to me.

Reply:
Considered this, but it would mean adding a new param `backend_found_failure` to `_raise_to_user` in addition to `verified_by`, which feels like putting business logic inside string/reporting logic. In this PR `_verified_by` no longer means 'verified by' though, it really means 'if we find a new failure, this backend is responsible', so maybe your proposal is better.